command/e2etest/automation_test.go

```diff
@@ -71,15 +70,18 @@ func TestPlanApplyInAutomation(t *testing.T) {
 		t.Fatalf("failed to read plan file: %s", err)
 	}
 
-	stateResources := plan.State.RootModule().Resources
-	diffResources := plan.Diff.RootModule().Resources
+	// stateResources := plan.Changes.Resources
```

Comment:
@apparentlymart - before I remove the `WIP` tag from this PR I wanted to see if you had any thoughts about this test. The `data.template_file.test` does not show up in the new plan, which I expected; is there anything else I should be checking for in this test?

Reply:
Honestly I don't really remember what this part of the test was about... the logic below where the plan's initial state contains the data resource and the planned changeset contains the resource _feels_ right for [this configuration](https://github.com/hashicorp/terraform/blob/4897c0554da5d07eb06981dbd4bfc5d9e3bc22a5/command/e2etest/test-fixtures/full-workflow-null/main.tf), because the `data` block doesn't contain anything that should be unknown during plan.

I forget whether we're expecting the already-resolved data resource to appear in the _changes_ in the new world, but whether it does or not isn't really important for this case -- it'd presumably be recorded as a no-op anyway, if it were recorded at all -- so I think we can just make this test match the apparent behavior for that particular part. The presence of `null_resource.test` in the plan is the important thing.

Reply:
🤔  Can I get the plan's initial state from this test? I don't have access to the command's context (that I know how to access, at least!), and the `plans.Plan` doesn't contain that information.